library/rsa.c

```diff
@@ -408,11 +408,11 @@ int mbedtls_rsa_export( const mbedtls_rsa_context *ctx,
 
     /* Check if key is private or public */
     is_priv =
-        mbedtls_mpi_cmp_int( &ctx->N, 0 ) != 0 &&
-        mbedtls_mpi_cmp_int( &ctx->P, 0 ) != 0 &&
-        mbedtls_mpi_cmp_int( &ctx->Q, 0 ) != 0 &&
-        mbedtls_mpi_cmp_int( &ctx->D, 0 ) != 0 &&
-        mbedtls_mpi_cmp_int( &ctx->E, 0 ) != 0;
+        ctx->N.p != NULL &&
+        ctx->P.p != NULL &&
+        ctx->Q.p != NULL &&
```

Comment:
This doesn't fix the bug of treating a key with e.g. d ≠ 0 but p=q=0 as public.

Reply:
Because there is no such bug. There are only two legal ways of importing a key:
- by using pkparse
- by using a sequence of calls to `rsa_import()` followed by a call to `rsa_complete()`.
Once you imported a key in one of these ways, the condition you mention can never happen.

Reply:
s/bug/lack of robustness/ then

Reply:
I'd still consider that quite orthogonal to the primary issue the PR is addressing, and I'd rather address it separately.

Reply:
I'm fine with leaving it for later. But I found it surprising that you wouldn't fix this since you were changing this code anyway.